src/Core/src/Test/ResultMockFactory.php

```diff
@@ -30,16 +30,26 @@ class ResultMockFactory
      *
      * @return Result|T
      */
-    public static function createFailing(string $class, int $code, ?string $message = null)
-    {
+    public static function createFailing(
+        string $class,
+        int $code,
+        ?string $message = null,
+        ?string $awsCode = null,
+        ?string $awsType = null
+    ) {
         if (Result::class !== $class) {
             $parent = get_parent_class($class);
             if (false === $parent || Result::class !== $parent) {
                 throw new \LogicException(sprintf('The "%s::%s" can only be used for classes that extend "%s"', __CLASS__, __METHOD__, Result::class));
             }
         }
 
-        $httpResponse = new SimpleMockedResponse(\json_encode(['message' => $message]), ['content-type' => 'application/json'], $code);
+        $awsTypeContent = [];
+        if (null !== $awsCode || null !== $awsType) {
+            $awsTypeContent['__type'] = \sprintf('%s#%s', (string) $awsType, (string) $awsCode);
+        }
+
+        $httpResponse = new SimpleMockedResponse(\json_encode(\array_merge(['message' => $message], $awsTypeContent)), ['content-type' => 'application/json'], $code);
         $client = new MockHttpClient($httpResponse);
```

Comment:
Couldn't we just remove typehint `string` fom message. and let user provide an array with the `__type` argument?

Reply:
@jderusse could you clarify what you mean by `removing string typehint from message`? 
Do I understand correctly that yours proposal is to make $message type mixed and check if its string, then add "message" to body and if its array - just pass it as response body?

Reply:
nvm, I prefer the last version of your PR ;-)